types.cc

```diff
@@ -1888,6 +1893,58 @@ data_value abstract_type::deserialize(bytes_view v) const {
     return visit(*this, deserialize_visitor{v});
 }
 
+int32_t compare_aux(const tuple_type_impl& t, bytes_view v1, bytes_view v2) {
+    // This is a slight modification of lexicographical_tri_compare:
+    // when the only difference between the tuples is that one of them has additional trailing nulls,
+    // we consider them equal. For example, in the following CQL scenario:
+    // 1. create type ut (a int);
+    // 2. create table cf (a int primary key, b frozen<ut>);
+    // 3. insert into cf (a, b) values (0, (0));
+    // 4. alter type ut add b int;
+    // 5. select * from cf where b = {a:0,b:null};
+    // the row with a = 0 should be returned, even though the value stored in the database is shorter
+    // (by one null) than the value given by the user.
+
```

Comment:
Maybe it's better to deserialize undersized UDTs to the full length. Then we kill the complexity then and there, and the rest of the code can deal with full-sized UDTs.

Reply:
By giving the tuple type to tuple_deserializing_iterator?